cmake/modules/AddRootBench.cmake

```diff
@@ -60,25 +78,42 @@ endfunction(RB_ADD_GBENCHMARK)
 
 
 #----------------------------------------------------------------------------
-# function RB_ADD_PYTESTBENCHMARK(file_name)
+# function RB_ADD_PYTESTBENCHMARK(<benchmark> filename)
 #
 # Dependency: pytest with pytest-benchmark and PyROOT modules
 #
 #----------------------------------------------------------------------------
-function(RB_ADD_PYTESTBENCHMARK file_name)
+function(RB_ADD_PYTESTBENCHMARK benchmark)
+  cmake_parse_arguments(ARG "" "LABEL" "SETUP;DOWNLOAD_DATAFILES" ${ARGN})
   if(ROOT_pyroot_experimental_FOUND OR ROOT_pyroot_FOUND)
-    set(ROOT_ENV ROOTSYS=${ROOTSYS}
-        PATH=${ROOTSYS}/bin:$ENV{PATH}
-        LD_LIBRARY_PATH=${ROOTSYS}/lib:$ENV{LD_LIBRARY_PATH}
-        PYTHONPATH=${ROOTSYS}/lib:$ENV{PYTHONPATH})
+    if(${ARG_LABEL} STREQUAL "long")
+      set(${TIMEOUT_VALUE} 1200)
+    elseif($ARG_LABEL STREQUAL "short")
+      set(${TIMEOUT_VALUE} 3600)
+    endif()
+    set(filename ${ARG_UNPARSED_ARGUMENTS})
+
     if(PYTEST_FOUND)
-      file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/${file_name}.py DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/${file_name}.py USE_SOURCE_PERMISSIONS)
-      ROOT_ADD_TEST(rootbench-${file_name}
-                COMMAND ${PYTHON_EXECUTABLE} -B -m pytest ${CMAKE_CURRENT_SOURCE_DIR}/${file_name}.py -v --csv rootbench-pytest-${file_name}.csv
-                ENVIRONMENT ${ROOT_ENV})
+      if(ARG_DOWNLOAD_DATAFILES)
+          RB_ADD_DOWNLOAD_FIXTURE(${benchmark} DOWNLOAD_DATAFILES ${ARG_DOWNLOAD_DATAFILES})
+      endif()
+
+      if(ARG_SETUP)
+         RB_ADD_SETUP_FIXTURE(${benchmark} SETUP ${ARG_SETUP})
+      endif()
+
+      file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/${filename} DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/${benchmark} USE_SOURCE_PERMISSIONS)
+
+      add_test(NAME rootbench-${benchmark}
+               COMMAND ${PYTHON_EXECUTABLE} -B -m pytest ${CMAKE_CURRENT_SOURCE_DIR}/${filename} -v --csv rootbench-pytest-${benchmark}.csv)
+      set_tests_properties(rootbench-${benchmark} PROPERTIES
+                           ENVIRONMENT "LD_LIBRARY_PATH=${ROOT_LIBRARY_DIR}:$ENV{LD_LIBRARY_PATH};PYTHONPATH=${ROOTSYS}/lib:$ENV{PYTHONPATH};RB_DATASETDIR=${RB_DATASETDIR}"
+                           TIMEOUT "${TIMEOUT_VALUE}" LABELS "${ARG_LABEL}" RUN_SERIAL TRUE
+                           FIXTURES_REQUIRED "setup-${benchmark};download-${benchmark}-datafiles")
+    else()
+      message(STATUS "ROOT was configured without PyROOT support. Python benchmarks will be disabled!")
```

Comment:
if I read correctly indentation of this message is now wrong and it's executed if `NOT PYTEST_FOUND`

Reply:
You are right, fixed!